gateway/ln-gateway/src/lib.rs

```diff
@@ -87,33 +86,10 @@ impl Gateway {
         decoders: ModuleDecoderRegistry,
         module_gens: ClientModuleGenRegistry,
         task_group: TaskGroup,
-    ) -> Self {
+    ) -> Result<Self> {
         // Create message channels for the webserver
         let (sender, receiver) = mpsc::channel::<GatewayRequest>(100);
 
-        // Source route hints form the LN node
-        let mut num_retries = 0;
-        let route_hints = loop {
-            let route_hints: Vec<RouteHint> = lnrpc
-                .routehints()
-                .await
-                .expect("Could not fetch route hints")
-                .try_into()
-                .expect("Could not parse route hints");
-
-            if !route_hints.is_empty() || num_retries == ROUTE_HINT_RETRIES {
-                break route_hints;
-            }
```

Comment:
It doesn't block *forever*, currently only 20s. The reason for this is that only online channels are returned (to avoid returning permanently offline/closed ones) and after a node comes online it may take a moment to get its channels up. Iirc this was a fix to an actual problem that happened infrequently and only in production (in tests envs channels come oneline super quickly).

The real fix would be re-fetching this list on every re-registration. Then we wouldn't mind as much there being no route hints for the first 5 minutes or so after restart.

Reply:
This makes sense. I didn't understand the offline channel part. I figured the only change would be if a new channel showed up, which seemed unlikely in such a short timespan.

Reply:
The plugin starts alongside CLN and right after restart all channels are offline because CLN didn't have time to get them all up. They will connect ion a few seconds typically if they aren't Tor-connected channels.